client/internal/wsreceiver_test.go

```diff
@@ -142,3 +149,18 @@ func TestDecodeMessage(t *testing.T) {
 		}
 	}
 }
+
+func TestWSReceiverReceiverLoop(t *testing.T) {
+	ctx := context.Background()
+	// Start a Server.
+	srv := StartMockServer(t)
+	defer srv.Close()
+	dialer := *websocket.DefaultDialer
+	conn, _, err := dialer.DialContext(ctx, "ws://"+srv.Endpoint, make(http.Header))
+	assert.NotNil(t, conn)
+	assert.NoError(t, err)
+	err = conn.Close()
```

Comment:
Is there any way that we can use wsClient here instead of create a websocket on our own? The [bug](https://github.com/open-telemetry/opamp-go/issues/163) is about when the wsClient is stopped, so this tests technically doesn't verified that, it tests something else that creates a similar situation but if wsClient implementation in the future changes this test will no longer be useful..